plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployCommandHandler.java

```diff
@@ -160,9 +161,10 @@ private void launchDeployJob(IProject project, Credential credential)
     errorStream.setColor(colorProvider.getColor(IDebugUIConstants.ID_STANDARD_ERROR_STREAM));
```

Comment:
Wouldn't this location be the better place to raise the console?

Reply:
It is doing so already. See a few lines above.

Reply:
Then why not just use the same approach as #2346 and use `setActivateOnWrite()`?

Reply:
Oh, that's the way we should do it. Thanks for the tip.

Reply:
I'd like to at least consult the user's preference as to whether they want the raise-on-write for stdout and stderr.  I personally loathe this behaviour and so have it turned off.

Reply:
There is one conceptual problem. As you can see in [this screenshot](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/2327#issuecomment-324944085), the preference you are referring to is under the hierarchy of _Run/Debug_ > _Console_. The settings there are meant for consoles from run/debug launch configuration. So, conceptually, applying the settings globally to every console is actually out of the intended scope (although it's not terribly unreasonable). Our console is custom-created programmatically, not from run/debug, so it feels a bit weird and surprising to find that the run/debug preferences apply to a console of a custom job. Likewise, that's the not only settings in that hierarchy. We don't honor, e.g., _Limit console output_. Should our custom console be committed to support all the existing/future preferences of run/debug (rhetorical question)? It's not difficult to check the preferences and apply it, but it irks me a bit because of this reason.

That said, to be conceptually correct, I think we should have our own preferences panel for this.

In any case, can we consider such support as enhancement to be implemented later? If you do not agree, I think it's better to we go back to the original PR to show the console just once more when the deploy starts, as a stop-gap.

Reply:
Run/Debug is the UI labels for the launch framework, and include consoles, etc.

I'm fine with this for now; if we can cast the process as an `IProcess` then [we'll get this for free](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/2140#issuecomment-326387162).

Reply:
> Run/Debug is the UI labels for the launch framework, and include consoles, etc.

Ah, I see.